debugger/src/debug_adapter.rs

```diff
@@ -933,46 +913,78 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
-            request,
-            match self
-                .variable_map
-                .clone()
-                .get(&(arguments.variables_reference))
+
+        let response = if let Some(debug_info) = core_data.debug_info {
+            // During the intial stack unwind operation, if we encounter certain types of pointers as children of complex variables, they will not be auto-expanded and included in the variable cache. Please refer to the `is_pointer` member of [probe_rs::debug::Variable] for more information. If this is the case, we will store the `stack_frame_registers` as part of the variable definition, so that we can  resolve the variable and add it to the cache before continuing.
+            // TODO: Use the DAP "Invalidated" event to refresh the variables for this stackframe. It will allow the UI to see updated compound values for pointer variables based on the newly resolved children.
+            if let Some(parent_variable) = debug_info
+                .variable_cache
+                .get_variable_by_key(arguments.variables_reference)
             {
-                Some(dap_variables) => {
-                    match arguments.filter {
-                        Some(filter) => {
-                            match filter.as_str() {
-                                // TODO: Use `probe_rs::Variables` for the `variable_map`, and then transform them here before serving them up.
-                                // That way we can actually track indexed versus named variables (The DAP protocol doesn't have Variable fields to do so).
-                                "indexed" => Ok(Some(VariablesResponseBody {
-                                    variables: dap_variables.clone(),
-                                })),
-                                "named" => Ok(Some(VariablesResponseBody {
-                                    variables: dap_variables.clone(),
-                                })),
-                                other => Err(DebuggerError::Other(anyhow!(
-                                    "ERROR: Received invalid variable filter: {}",
-                                    other
-                                ))),
-                            }
+                if parent_variable.referenced_node_offset.is_some() {
+                    debug_info
+                        .cache_referenced_variables(&mut core_data.target_core, parent_variable)?;
+                }
+            }
+
+            let dap_variables: Vec<Variable> = debug_info
+                .variable_cache
+                .get_children(arguments.variables_reference)?
+                .iter()
+                // Filter out requested children, then map them as DAP variables
+                .filter(|variable| match &arguments.filter {
+                    Some(filter) => match filter.as_str() {
+                        "indexed" => variable.name.starts_with("__"),
+                        "named" => !variable.name.starts_with("__"),
```

Comment:
This is how the names are encoded in the DWARF (I can't find any documentation pointing to why/when, but the DW_AT_name values just show up like that). We propogate the convention when creating the children of arrays for consistency with all the other variable names. I feel like this is right the place (when DAP wants to know if a variable is indexed or not), to do the interpretation. There really is no other time that the indexed/non-indexed nature has any value. 

Having said that, interpreting the value as indexed because it starts with __ is not strictly speaking correct. It really should be that it starts with an __ immediately followed by a digit.

Reply:
Done. We now have a `Variable::is_indexed()`